lib/link.ts

```diff
@@ -208,14 +220,17 @@ export abstract class Link extends Entity {
   }
 
   /**
-   * Closes the underlying amqp link and session in rhea if open. Also removes all the event
-   * handlers added in the rhea-promise library on the link and it's session
-   * @return {Promise<void>} Promise<void>
+   * Closes the underlying amqp link and optionally the session as well in rhea if open.
+   * Also removes all the event handlers added in the rhea-promise library on the link
+   * and optionally it's session.
+   * @returns Promise<void>
    * - **Resolves** the promise when rhea emits the "sender_close" | "receiver_close" event.
    * - **Rejects** the promise with an AmqpError when rhea emits the
    * "sender_error" | "receiver_error" event while trying to close the amqp link.
    */
-  async close(): Promise<void> {
+  async close(options?: LinkCloseOptions): Promise<void> {
```

Comment:
@GaikwadPratik the capability to not close the underlying session may be useful to you, if you are planning to create all the links (sender | receiver) on the same session.

Reply:
@amarzavery 
I will try this pull request shortly. Although this looks cool...